modules/client/src/addressList/internal/client/methods.ts

```diff
@@ -70,18 +74,56 @@ export class ClientAddressListMethods extends ClientCore
       throw new Error("A web3 provider is needed");
     }
 
-    // TODO: Implement
-    await delay(1000);
+    const addresslistContract = AllowlistVoting__factory.connect(
+      _params.pluginAddress,
+      signer,
+    );
+
+    let cid = "";
+    try {
+      cid = await this.ipfs.add(JSON.stringify(_params.metadata));
+    } catch {
+      throw new Error("Could not pin the metadata on IPFS");
+    }
+
+    const tx = await addresslistContract.createVote(
+      toUtf8Bytes(cid),
+      _params.actions || [],
+      Math.round(_params.startDate?.getTime() || 0 / 1000),
+      Math.round(_params.endDate?.getTime() || 0 / 1000),
+      _params.executeOnPass || false,
+      _params.creatorVote || 0,
+    );
+
     yield {
       key: ProposalCreationSteps.CREATING,
-      txHash:
-        "0x0123456789012345678901234567890123456789012345678901234567890123",
+      txHash: tx.hash,
     };
-    await delay(3000);
+
+    const receipt = await tx.wait();
+    const addresslistContractInterface = AllowlistVoting__factory
+      .createInterface();
+    const log = receipt.logs.find(
+      (log) =>
+        log.topics[0] ===
+          id(
+            addresslistContractInterface.getEvent("VoteCreated").format(
+              "sighash",
+            ),
+          ),
+    );
+    if (!log) {
+      throw new Error("Failed to create proposal");
+    }
+
+    const parsedLog = addresslistContractInterface.parseLog(log);
+    if (!parsedLog.args["voteId"]) {
+      throw new Error("Failed to create proposal");
+    }
+
     yield {
       key: ProposalCreationSteps.DONE,
-      proposalId:
-        "0x0123456789012345678901234567890123456789012345678901234567890123",
+      proposalId: hexZeroPad(parsedLog.args["voteId"].toHexString(), 32),
```

Comment:
Also below (can voteId have a null value?)

Reply:
we have a check on line 116